### ISSUE TITLE
training loss goes to nan

### ISSUE THREAD

**Issue by hachreak** [kevinraymond/wave-network#1]
Dear @kevinraymond ,
I was trying to use your wave-network model implementation on a image training task.
The training looks a bit unstable because at some point the loss goes to nan and the training fails.
Could you help me to understand what it is going bad?  Thank you very much.

Best,
Leo

**Comment by kevinraymond** [comment]
Wave Network is really made for NLP tasks - what are you trying to do with images?

It's interesting to think about though ... I suppose it could be adapted to use image features and pixels for the signals, but that would require refactoring everything to work with image data instead of text. Could be a fun research project though!

**Comment by hachreak** [comment]
Hi @kevinraymond ,
thank you to quickly replying on my message.
I am working on Single Image Super-Resolution task, to upscale a single low-res images to a high quality version.
I think it would be interesting to apply wave-net to images.

Actually, I simply removed the [embedding creation](https://github.com/kevinraymond/wave-network/blob/main/wave_network.py#L118), the [pooling and classification layers](https://github.com/kevinraymond/wave-network/blob/main/wave_network.py#L135-L136) and use it instead of a mamba/transformer layer.

Usually a mamba/transformer layer for images takes in input [B, L, C], where B is the batch size, L number of tokens, C token size.
For this reason, the simplest way to use Wave Network has been removing the embedding layer.
I don't know if it could compromise the performance.
If you are interested, I can share a demo :smile: 

Talking about research, recently a paper called WaveMLP [1] has been proposed, and I would say that the idea sound similar to wave network.
Have you heard about that?

[1] Tang, Yehui, et al. "An image patch is a wave: Phase-aware vision mlp." Proceedings of the IEEE/CVF conference on computer vision and pattern recognition. 2022.

**Comment by kevinraymond** [comment]
Interesting, thanks for dropping the reference - I will try to take a look!

I haven't been able to spend more time looking into this but it sounds like you've got something (maybe) working for your needs. I also have no idea about performance being affected.

So many papers and so little time ...

**Comment by kevinraymond** [comment]
It has been a while but I made some updates that should eliminate the NaN issues in any training. Coming back around since this is related.
